### PR TITLE
fix(deps): bump html-proxy-middleware to latest version

### PR DESCRIPTION
**Pull request by vtereshyn**
This PR fixes an audit issue.

Also, in the scope of this pull request, I ran `npm audit fix` to resolve some audit issues. That's why lots of changes in `package-lock.json`

### CODE REVIEW

**Comment by vtereshyn** [comment]
I am not sure what should I do with commit message

File: package.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
     "del": "^4.1.1",
     "express": "^4.17.1",
     "html-entities": "^1.3.1",
-    "http-proxy-middleware": "0.19.1",
+    "http-proxy-middleware": "1.0.4",
```
Comment (chimurai) [inline package.json:48]: How you use `require` changed in http-proxy-middleware version 1.0.0. Looks like you didn't update those statements.

https://github.com/chimurai/http-proxy-middleware/releases/tag/v1.0.0

v0.x.x with implicit require

```js
const proxy = require('http-proxy-middleware');
```

v1.0.0 with explicit require

```js
const { createProxyMiddleware } = require('http-proxy-middleware');
```

Comment (hughrawlinson) [reply package.json:48]: Wondering why this package's version is locked - does need to stay locked?

Comment (vtereshyn) [reply package.json:48]: probably, I changed locked version to the latest version and locked it.

Comment (vtereshyn) [reply package.json]: alright, I removed version locking and fixed `http-proxy-middleware` usage

**Comment by vtereshyn** [comment]
Not sure what's wrong with tests and other stuff, but I ran tests, lints on my local machine several times and everything looks good. 

https://prnt.sc/sjgxzd

**Comment by evilebottnawi** [comment]
Can't be merged for `v3` because it is breaking change, will be shipped in `v4`

**Comment by vtereshyn** [comment]
> Can't be merged for `v3` because it is breaking change, will be shipped in

what should we do with `audit` issue? our checks don't pass. I know that a lot of repositories (almost 4 million) use `webpack-dev-server`.

**Comment by evilebottnawi** [comment]
@vtereshyn temporary you can don't audit dev dependencies

**Comment by vtereshyn** [comment]
@evilebottnawi I am not sure if security managers will like that way :)

**Comment by vtereshyn** [comment]
@evilebottnawi sorry, just wondering, which breaking changes are we talking about? I mean, I've checked this pull request with my application and everything looks good. But of course, you know better. thank you in advance

**Comment by koltyakov** [comment]
I assume it's about Node.js version and the fact that http-proxy-middleware v1 supports Node v8 and higher.

**Comment by evilebottnawi** [comment]
@vtereshyn https://github.com/webpack/webpack-dev-server/pull/2616#issuecomment-630749121

**Comment by lsiler-mdsol** [comment]
@evilebottnawi thanks for the info! when can we expect v4 to be released?

**Comment by evilebottnawi** [comment]
@lsiler-mdsol I am working on it, sorry can't provide ETA, because I am working only one, but I try to do it ASAP

**Comment by chimurai** [comment]
Just published `http-proxy-middleware@0.19.2`

Hopefully this'll ease the process to patch the current version of WDS without rushing to release WDS@4.0

**Comment by hughrawlinson** [comment]
What do you think about releasing wds@3.11.1 now that there's a patch fix of `http-proxy-middleware`? That would get around having to block until wds@4.

**Comment by varora1406** [comment]
i have submitted a PR with just patch of `http-proxy-middleware` Can we get #2665 going?

**Comment by evilebottnawi** [comment]
Done in v4, for v3 we can backport only https://github.com/webpack/webpack-dev-server/pull/2665

**Pull request closed without merging**